### ISSUE TITLE
memory detection fails on OpenBSD

### ISSUE THREAD

**Issue by gperciva** [Tarsnap/scrypt#226]
We currently ignore `RLIMIT_DATA` if we `HAVE_MMAP`:
https://github.com/Tarsnap/scrypt/commit/2b478e7ce590a6564d89417ddad4f3dd4bbdcaf6

This breaks the memory-detection on OpenBSD.   I added some `zzz` lines, and compiled with `-DDEBUG`, and we see:

```
openbsd-64$ ulimit -d 123456
openbsd-64$ ./scrypt dec -v foo.enc foo                                        
Please enter passphrase: 
zzz             RLIMIT_DATA:    126418944
zzz                     (the above line is for debugging only
zzz             RLIMIT_RSS:     2055184384
Memory limits are:
        usermem:        18446744073709551615
        memsize:        18446744073709551615
        sysinfo:        18446744073709551615
        rlimit:         2055184384
        sysconf:        2130640896
zzz smallest:           2055184384
zzz after frac:         1027592192
zzz after specified:    1027592192
Allowing up to 1027592192 memory to be used
Clock resolution is 0.01
17408 salsa20/8 cores performed in 0.010059 seconds
Parameters used: N = 131072; r = 8; p = 4;
    This requires at least 134 MB bytes of memory (1.0 GB available),
    and will take approximately 1.2 seconds (limit: 300.0 seconds).
scrypt: ../main.c, 252
scrypt: Error computing derived key: Cannot allocate memory
openbsd-64$ 
```

TL;DR: scrypt thinks it has enough memory available, but it doesn't.

**Comment by gperciva** [comment]
Original report from Ted Unangst here: http://mail.tarsnap.com/scrypt/msg00263.html

**Comment by gperciva** [comment]
Resolved by #238.
